bin/benchwrapper.js

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const grpc = require('grpc');
+const protoLoader = require('@grpc/proto-loader');
+const {Spanner} = require('../build/src');
+
+const argv = require('yargs')
+  .option('port', {
+    description: 'The port that the Node.js benchwrapper should run on.',
+    type: 'number',
+    demand: true,
+  })
+  .parse();
+
+const PROTO_PATH = __dirname + '/spanner.proto';
+const packageDefinition = protoLoader.loadSync(PROTO_PATH, {
+  keepCase: true,
+  longs: String,
+  enums: String,
+  defaults: true,
+  oneofs: true,
+});
+const protoDescriptor = grpc.loadPackageDefinition(packageDefinition);
+const spannerBenchWrapper = protoDescriptor.spanner_bench;
+
+// The benchwrapper should only be executed against an emulator.
+if (!process.env.SPANNER_EMULATOR_HOST) {
+  throw new Error(
+    'This benchmarking server only works when connected to an emulator. Please set SPANNER_EMULATOR_HOST.'
+  );
+}
+// This will connect the Spanner client to an emulator, as SPANNER_EMULATOR_HOST has been set.
+const spannerClient = new Spanner();
+
+// Implementation of SpannerBenchWrapper.Read method.
+function Read(call, callback) {
+  const instance = spannerClient.instance('someinstance');
+  const database = instance.database('somedatabase');
```

Comment:
I assume that spannerClient will cache instance, and the instance will cache database. So every time we call this, it will return the same database object.

Reply:
Yes, both are cached and the same instances are returned.